### DISCUSSION TITLE
Feature Request:  JSON support for `workerd serve`

### DISCUSSION THREAD

**Discussion by isaksky** [cloudflare/workerd#859] [Ideas]
There are currently two ways to provide a configuration to `workerd serve`:

- A capnproto text file (e.g., `workerd serve my-config.capnp`)
- Binary encoded capnproto (with the `--binary` flag)

For driving workerd with another program in ecosystems with poor capnproto support, it would be nice to be able to provide configuration via JSON.

Looks like the ability to translate back and forth from JSON is available in capnproto here:

https://github.com/capnproto/capnproto/blob/master/c++/src/capnp/compat/json.h

**Comment by jasnell** [comment]
I'm going to convert this into a discussion as we generally try to keep the issue tracker for immediately actionable issues (i.e. bugs). This does seem like a reasonable request but it's probably not something that we would be able to prioritize quickly. We would consider accepting a PR tho! If you're feeling brave and intrepid enough to dig into the details of how it could be implemented :-)
